### PR TITLE
torch.where changes made on 1.3.1 but not on master

### PR DESCRIPTION
**Pull request by nairbv**
Fixes for torch.where (https://github.com/pytorch/pytorch/pull/29078) were made on the 1.3.1 branch but not on master at the time of the 1.4.0 branch cut.

This PR cherry-picks those two changes against the v1.4.0 branch.

Original issue: https://github.com/pytorch/pytorch/issues/28709
1.3.1 PR: https://github.com/pytorch/pytorch/pull/29078
cherry-picked PR to master: https://github.com/pytorch/pytorch/pull/30662

### CODE REVIEW

File: aten/src/ATen/native/Loss.cpp [hunk 1]
```diff
@@ -77,8 +77,8 @@ Tensor margin_ranking_loss(const Tensor& input1, const Tensor& input2, const Ten
 }
 
 Tensor kl_div(const Tensor& input, const Tensor& target, int64_t reduction) {
-  auto zeros = at::zeros_like(target, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
   auto output_pos = target * (at::log(target) - input);
+  auto zeros = at::zeros_like(output_pos, LEGACY_CONTIGUOUS_MEMORY_FORMAT);
```
Comment (vadimkantorov) [inline aten/src/ATen/native/Loss.cpp]: If torch.where accepted scalars / python scalars, we would not need these extra allocation https://github.com/pytorch/pytorch/issues/9190

**Review by zou3519 — APPROVED** [review]
